### DISCUSSION TITLE
Adding `align-items` for `stack`

### DISCUSSION THREAD

**Discussion by duskmoon314** [typst/typst#1325] [Ideas]
Currently, `stack` only has two parameters: `dir` for direction and `spacing`. I have an idea to add a feature like `align-items` of CSS to control how contents are aligned in the cross direction.

## My expectation

Let's say we add a new parameter `align-items` with 4 possible values: `start`, `end`, `center`, and `stretch`, just like the one in CSS. And we can use it like this:

### `align-items: start`
```typst
#stack(
  dir: ltr,
  align-items: start,
  // some contents with height 1 3 1, represented by `x` in the following contents
)
```
This will lead to something like this:
```
// start
x  x  x
   x
   x
// end
```

### `align-items: end`
``` typst
#stack(
  dir: ltr,
  align-items: end,
  // some contents as above
)
```
This will lead to something like this:
```
// start
   x
   x
x  x  x
// end
```

### `align-items: center`
```typst
#stack(
  dir: ltr,
  align-items: center,
  // some contents as above
)
```
This will lead to something like this:
```
   x
x  x  x
   x
```

### `align-items: stretch`
```typst
#stack(
  dir: ltr,
  align-items: center,
  // some contents as above
)
```
This will lead to something like this:
```
x  x  x
x  x  x
x  x  x
```

## A workaround

We can place `align(top/horizon/bottom, content)` around every content in the `stack`, which is a little bothering.

## REF

https://developer.mozilla.org/en-US/docs/Web/CSS/align-items

**Comment by laurmaedje** [comment]
You can also do an `align` set rule outside of the stack.

**Reply by duskmoon314** [reply]
I have yet to notice this usage. But after testing, it is not quite the same. The `align` outside controls the whole `stack` rather than only the contents inside the `stack`.

Suppose we want to place a `stack` at the top left corner of a `box`, and the contents inside are aligned with `horizon + center`. Unfortunately, only one `align` outside of `stack` seems unable to achieve this. A workaround is to use a `block` to control where the `stack` is placed.

The code I tried is:
```typst
// block is used to represent an area to place the stack
#block(
    stroke: 1pt,
    inset: 1mm,
    width: 130mm,
    height: 20mm,
    align(
        // align here controls the whole stack
        // and the contents of the stack
        top + left,
        stack(
            dir: ltr,
            spacing: 2mm,
            rect(
                width: 10mm,
                height: 10mm,
                "1"
            ),
            rect(
                width: 10mm,
                height: 5mm,
                "2"
            )
        )
    )
)
```

I may have missed some usage. Please share your ideas. 😃
